### PR TITLE
[Platform]: Search results page option updated

### PR DESCRIPTION
**Pull request by chinmehta**
# [Platform]: Search results page option updated.

## List options of search results updated on user input instead of result from debounced input. 

**Issue:** # [Issue-2902](https://github.com/opentargets/issues/issues/2902)
**Deploy preview:** [Preview](https://deploy-preview-105--ot-platform.netlify.app/)

## Type of change

Please delete options that are not relevant.

* Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

* Test A: If a term is entered in search and hit enter within 300ms, the resulting page will show search input with last input value registered(debounced). 
* Test B: Input a term and hit enter within 300ms.

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by LucaFumis — APPROVED** [review]
Thank you for fixing that @chinmehta

File: packages/ui/src/Search/SearchInput.tsx [hunk 1]
```diff
@@ -80,11 +78,11 @@ function SearchInput({
...
-  useEffect(() => {
-    if (inputRef.current) {
-      inputRef.current.value = inputValue;
-    }
-  }, [inputValue]);
+  // useEffect(() => {
+  //   if (inputRef.current) {
+  //     inputRef.current.value = inputValue;
+  //   }
+  // }, [inputValue]);
```
Comment (carcruz) [inline packages/ui/src/Search/SearchInput.tsx:85]: Remove commented code

Comment (chinmehta) [reply packages/ui/src/Search/SearchInput.tsx]: done

File: packages/ui/src/AutocompleteSearch.tsx [hunk 2]
```diff
@@ -129,28 +131,24 @@ export default function AutocompleteSearch({
...
   const onClose = () => {
     setLoading(false);
     setInputValue("");
+    setOpen(false);
     closeModal();
   };
 
-  const handleSelectOption = (e, option) => {
+  const handleSelectOption = (
+    event: React.ChangeEvent<{}>,
```
Comment (carcruz) [inline packages/ui/src/AutocompleteSearch.tsx:142]: Import `ChangeEvent` from react.

```
import { ChangeEvent } from 'react';
```

Comment (chinmehta) [reply packages/ui/src/AutocompleteSearch.tsx]: done

**Review by carcruz — APPROVED** [review]
Good to go

**Pull request merged**